core/src/main/scala/kafka/admin/DeleteRecordsCommand.scala

```diff
@@ -22,31 +22,62 @@ import java.util.Properties
 
 import kafka.admin.AdminClient.DeleteRecordsResult
 import kafka.common.AdminCommandFailedException
-import kafka.utils.{CoreUtils, Json, CommandLineUtils}
+import kafka.utils.{CommandLineUtils, CoreUtils, Json}
 import org.apache.kafka.common.TopicPartition
 import org.apache.kafka.common.utils.Utils
 import org.apache.kafka.clients.CommonClientConfigs
 import joptsimple._
+import kafka.utils.json.JsonValue
 
 /**
  * A command for delete records of the given partitions down to the specified offset.
  */
 object DeleteRecordsCommand {
 
+  private[admin] val EarliestVersion = 1
+  private[admin] val supportedVersions = List(EarliestVersion)
+
   def main(args: Array[String]): Unit = {
     execute(args, System.out)
   }
 
   def parseOffsetJsonStringWithoutDedup(jsonData: String): Seq[(TopicPartition, Long)] = {
-    Json.parseFull(jsonData).toSeq.flatMap { js =>
-      js.asJsonObject.get("partitions").toSeq.flatMap { partitionsJs =>
-        partitionsJs.asJsonArray.iterator.map(_.asJsonObject).map { partitionJs =>
-          val topic = partitionJs("topic").to[String]
-          val partition = partitionJs("partition").to[Int]
-          val offset = partitionJs("offset").to[Long]
-          new TopicPartition(topic, partition) -> offset
-        }.toBuffer
+    Json.parseFull(jsonData) match {
+      case Some(js) => {
+
+        val version = js.asJsonObject.get("version") match {
+          case Some(jsonValue) => jsonValue.to[Int]
+          case None => EarliestVersion
+        }
+        if (!supportedVersions.contains(version))
+          throw new AdminOperationException(s"Not supported version field value $version")
+
+        parseJsonData(version, js)
+      }
+      case None => throw new AdminOperationException("Delete records data is empty")
```

Comment:
I am not sure we need to throw exception here. Can we keep the existing behavior and return an empty sequence in this case?

Reply:
According to the Java doc of parseFull(), it seems that parseFull() returns None only if the input string is not valid JSON, right? If so, one more minor comment is that the error message can be something like `the input string is not a valid json`.

Reply:
Agree.